src/core/Xlang.mli

```diff
@@ -37,6 +41,24 @@ val to_langs : t -> Lang.t list
 (* raises an exception with error message *)
 val lang_of_opt_xlang_exn : t option -> Lang.t
 
+(*
+   Determine whether a single analyzer exist in an set of
+   analyzers. If the first analyzer is more than one language, it's
+   an error and the result is always false (it could be an exception but
+   this is safer).
+
+   This is normally used for determining if a target that's
+   expected to be analyzable with one analyzer can be analyzed with
+   the rule's listed analyzers ("languages").
+
+   Examples:
+   - 'LRegex' is not compatible with 'L (JavaScript, [])'
+   - 'LRegex' is compatible only with 'LRegex'
+   - 'L (JavaScript, [])' is compatible with 'L (TypeScript, [JavaScript])'
+   - '~require:(L (TypeScript, [JavaScript]))' does not make sense.
+*)
+val is_compatible : require:t -> provide:t -> bool
```

Comment:
So the target language would the `require` whereas the rule languages would be the `provide` ?

Reply:
That's my understanding, but it took me a few reads to be confident

Reply:
yes